### PR TITLE
CI / CD

### PR DESCRIPTION
**Pull request by lukasmerten**
This PR add two new workflows, that are the starting point of continuous integration and deployment of the CRPropa-Data repository.

The first workflow (create-tarball), runs the calc_all.py file, which creates the default CRPropa data tarball and compresses it into a tar file. The checksum is calculated and stored separatly.

The second workflow, which is only manually triggered, uploads the tarball to sciebo, where the data files are hosted for CRPropa installation download.

After that the corresponding lines in CRPropa's CMakeLists.txt have to be updated.

### CODE REVIEW

**Comment by lukasmerten** [comment]
Note, that I needed to remove the crpropa dependency in the some of the calculation files, as we would be otherwise requiered to install crpropa on the github action server, when the tar ball should be created. 

As we were using mainly crpropa units, I have exported that functionality to a separate python module.

**Comment by rafaelab** [comment]
Thanks for the effort, @lukasmerten.
Unfortunately, I won't be able to review this in the next 2 weeks. 
Maybe someone else can take a look and test it before the 3.2.1 release.

File: calc_all.py [hunk 1]
```diff
@@ -217,7 +218,8 @@ def calc_checksum():
     print("#"*50)
     print("Calculating the checksum.")
     t1 = time.time()
-    datestr = "-".join([str(x) for x in time.localtime()[:3]])
+    # datestr format YYYY-MM-DD
+    datestr = "-".join([str(x).zfill(2) for x in time.localtime()[:3]])
```
Comment (JulienDoerner) [inline calc_all.py:222]: The `datestr` is used several times in the code and always re-evaluated. This should only be done once at the beginning.

File: calc_decay.py [hunk 2]
```diff
@@ -1,14 +1,47 @@
...
 import gitHelp as gh
 import os
+from units import mass_electron, c_squared, c_light, h_planck, keV, amu
 
 cdir = os.path.split(__file__)[0]
 
 # Script to preprocess the nuclear decay data table from the BNL NuDat2 database
 # Decay Search: http://www.nndc.bnl.gov/nudat2/indx_sigma.jsp, output: formatted file --> decay_NuDat2.txt
 # Decay Radiation Search: gamma_NuDat2.txt: http://www.nndc.bnl.gov/nudat2/indx_dec.jsp --> gamma_NuDat2.txt
+#
```
Comment (JulienDoerner) [inline calc_decay.py:12]: remove empty comment line

File: calc_decay.py [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
+    """Class to provide tabulated nuclear masses
+    
+    This function reimplements the basic functionality of
+    CRPropa's particleMass module see here:
+    https://github.com/CRPropa/CRPropa3/blob/master/include/crpropa/ParticleMass.h
+    """
+
+    def __init__(self):
+        try:
+            self.massTable = np.loadtxt('data/nuclear_mass.txt', usecols=(2))
```
Comment (JulienDoerner) [inline calc_decay.py:23]: change to global path of the file.

File: calc_decay.py [hunk 4]
```diff
@@ -1,14 +1,47 @@
...
+        except FileNotFoundError:
+            print("The file 'data/nuclear_mass.txt' was not found.")
+            print("Run the script calc_mass.py and try again.")
+
+    def getMass(self, id: int) -> float:
+        """Helper function to return tabulated nuclear masses"""
```
Comment (JulienDoerner) [inline calc_decay.py:29]: comment that this ID is not the typical PID in CRPropa.

File: calc_decay.py [hunk 5]
```diff
@@ -294,24 +329,24 @@ def __str__(self):
                 continue
 
             A  = Z+N
-            m1 = crp.nuclearMass(A, Z)
-            m2 = crp.nuclearMass(A, Z-1)
-            dm = (m1 - m2) * crp.c_squared
+            m1 = nucMass.nuclearMass(A, Z)
+            m2 = nucMass.nuclearMass(A, Z-1)
```
Comment (JulienDoerner) [inline calc_decay.py:333]: space around operator: `Z - 1`

File: calc_decay.py [hunk 6]
```diff
@@ -294,24 +329,24 @@ def __str__(self):
...
+                d.tau = np.inf
                 continue
 
-            f = lambda E: E * sqrt(E**2 - Qe**2) * (dm - E)**2
+            f = lambda E: E * np.sqrt(E**2 - Qe**2) * (dm - E)**2
             I, Ierr = quad(f, Qe, dm)
 
             # ratio tau_beta+ / tau_ec
-            f = pi**2 / 2 * (Z/a0*hbar_c)**3 * Qec**2 / I
+            f = np.pi**2 / 2 * (Z/a0*hbar_c)**3 * Qec**2 / I
```
Comment (JulienDoerner) [inline calc_decay.py:349]: spaces for opperator.

File: .gitignore [hunk 7]
```diff
@@ -1,4 +1,11 @@
 *.pyc
+*.tar.gz
+*.tar.gz*
 
 plots/
 temp/
+
+data/
+!data/CustomPhotonField/
+!data/dint/
```
Comment (JulienDoerner) [inline .gitignore:10]: DINT and EleCa are removed from CRPropa. Also remove it here.

**Review by JulienDoerner — CHANGES_REQUESTED** [review]
In general a realy good feature. 
I think the only thing that has to be cross checked is, who has the right to upload the data. I would guess due to security reasons this workflow sould not be triggered by users.

File: calc_all.py (hunk 1 above)
Comment (lukasmerten) [reply calc_all.py:222]: Done

File: calc_decay.py (hunk 2 above)
Comment (lukasmerten) [reply calc_decay.py:12]: Done

File: calc_decay.py (hunk 3 above)
Comment (lukasmerten) [reply calc_decay.py:23]: Done.

File: .gitignore (hunk 7 above)
Comment (lukasmerten) [reply .gitignore:10]: Done. Also removed the dint and eleca data directories.

File: calc_decay.py (hunk 6 above)
Comment (lukasmerten) [reply calc_decay.py:349]: Done

File: calc_decay.py (hunk 5 above)
Comment (lukasmerten) [reply calc_decay.py:333]: Done

File: calc_decay.py (hunk 4 above)
Comment (lukasmerten) [reply calc_decay.py:29]: Done

**Review by JulienDoerner — APPROVED** [review]

**Pull request merged**